pyxem/utils/correlation_utils.py

```diff
@@ -114,5 +114,41 @@ def _power(z, axis=0, mask=None, wrap=True, normalize=True):
         ).real
 
 
+def _pearson_correlation(z, mask=None):
+    """
+    Calculate Pearson cross-correlation of the image with itself
+    after rotation as a function of rotation
+
+     Parameters
+    ----------
+    z: np.array
+        Input image in 2D array
+    mask: np.array
+        A boolean mask to be applied.
+
+    Returns
+    -------
+    a: np.array
+        Pearson correlation of the input image
+
+    """
+    if mask is not None:
+        m = np.array(mask, dtype=bool)
+        mask_bool = ~m
```

Comment:
could you explain this to me?

Reply:
Yes. The discrete Fourier transform have a scaling factor of 1/N that needs to be normalized, where N is the length of the input wave. If there is no mask on the signal array, this is fairly easy as N is just the size of the signal in azimuthal axis (length of the row). That is the `z_length` in the `else` body. If there is a mask, this gets complicated because N is a variable that depends on the shift when you to the autocorrelation. It is essentially the number of unmasked pixels at different shift, i.e. the autocorrelation of the mask array.

This is what the `if` statement is doing, first four lines calculate the N `n_unmasked`. The `n_unmasked[n_unmasked < 1] = 1` line avoids dividing by zero for completely masked rows. And `z[m]` is just a easy way to manually apply the mask before sending it through Fourier transformation. Hope this is clear enough.

Reply:
yup, that helps a lot, I think `mask_bool` might be unused though?

Reply:
This seems okay, still a bit unsure why the inversion (ie `mask ---> mask_bool`) happens, but I think we can leave that as it is for now

Reply:
I think the logic of mask array in hyperspy is True for masked pixels, False for unmasked pixels (which is also pretty intuitive). So in this method, to calculate the number of unmasked pixels, we need to invert the boolean mask array.

Reply:
ah, okay, pyxem code tends to work the other way round. Probably should clear that up in the documentation...